tests/Schema/TableTest.php

```diff
@@ -610,6 +610,21 @@ public function testFullQualifiedTableName(): void
         self::assertEquals('other.test', $table->getFullQualifiedName('other'));
     }
 
+    public function testTableNameWithDatabaseAndSchema(): void
+    {
+        $table = new Table('foo.bar.test');
+        self::assertEquals('foo.bar.test', $table->getName());
+        self::assertEquals('foo.bar', $table->getNamespaceName());
```

Comment:
Or the dot could belong to the actual object name: https://github.com/doctrine/dbal/pull/5132.